src/Core/Usecase/HXL/SendHXLUsecase.php

```diff
@@ -0,0 +1,129 @@
+{
+    use Usecase\Concerns\IdentifyRecords;
+    use AuthorizerTrait; // ? do we need this here?
+    protected $metadataRepository;
+    protected $userSettingRepository;
+    protected $exportJobRepository;
+    protected $jobID;
+    protected $hdxInterface;
+
+        // @TODO: fetch these from the user settings when that exists
+    protected $userSettings = [
+        'hdx_server' => 'http://192.168.33.60:5000',
+        'user_key' => 'e0371305-e830-469f-adce-56f9ff211157'
+    ];
+
+    public function setExportJobRepository(ExportJobRepository $repo)
+    {
+        $this->exportJobRepository = $repo;
+    }
+
+    public function setUserHXLSettingsRepository(UserSettingRepository $repo)
+    {
+        $this->userSettingRepository = $repo;
+    }
+
+    public function setHXLMetadataRepository(HXLMetadataRepository $repo)
+    {
+        $this->metadataRepository = $repo;
+    }
+
+    public function setJobID($jobID)
+    {
+        $this->jobID = $jobID;
+    }
+
+    public function interact()
+    {
+        // get job by job_id
+        $job = $this->exportJobRepository->get($this->getIdentifier('job_id'));
+        // get user settings by user id
+        $user_settings = $this->userSettingRepository->getByUser($job->user_id);
+        // setup hdx interface
+        $this->setHDXInterface($user_settings);
+        // get metadata by job id
+        $metadata = $this->metadataRepository->getByJobId($this->getIdentifier('job_id'));
+        // check if the dataset exists to decide if we update or create one
+        $existing_dataset_id = $this->hdxInterface->getDatasetIDByTitle($metadata->dataset_title);
+        // TODO grab the tags assigned by the user to add them to the dataset's resource
+        // TODO Add resource creation
+        if (!!$existing_dataset_id) {
+            $updated_job = $this->updateDataset($metadata, $job);
+        } else {
+            $updated_job = $this->createDataset($metadata, $job);
+        }
+        return $this->formatter->__invoke($updated_job);
+    }
+
+    private function updateDataset($metadata, $job)
+    {
+        $results = $this->hdxInterface->updateHDXDatasetRecord($metadata->asArray());
+        if (isset($results['error'])) {
+            $job->setState(['status' => 'FAILED']);
+        } else {
+            $job->setState(['status' => 'SUCCESS']);
+        }
+        $this->exportJobRepository->update($job);
+        return $job;
+    }
+
+    private function createDataset($metadata, $job)
+    {
+        $results = $this->hdxInterface->createHDXDatasetRecord($metadata->asArray());
+        if (isset($results['error'])) {
+            $job->setState(['status' => 'FAILED']);
+        } else {
+            $job->setState(['status' => 'SUCCESS']);
+        }
+        $this->exportJobRepository->update($job);
+        return $job;
+    }
+
+    private function setHDXInterface($user_settings)
+    {
+        $this->hdxInterface = new HDXInterface(
```

Comment:
FIXME should be using $user_settings->config_value since it receives the config key with api_key in it

adding this in 30